### PR TITLE
minor: update 'field_input.ts' file

### LINKED ISSUE #7253: Update `widgetCreate_`  in `/core/field_input.ts` to return `HTMLInputElement` instead of `HTMLElement`
🚧⚠️🚧 This issue is being saved for **Grace Hopper Open Source Day**. 🚧⚠️🚧 An event we're running to help girls and non-binary technologists make their first contributions to open source! Please see our list of [help wanted issues](https://github.com/google/blockly/issues?page=1&q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+-label%3A%22ghc-osd%22) if you'd like to contribute! 

---

### Check for duplicates

- [X] I have searched for similar issues before opening a new one.

### Problem

This is a minor note, though in `/core/field_input.ts` we always know that the type info is of type `HTMLInputElement`, which is an `HTMLElement` but with more input-specific stuff. The proper type info should be returned so that input-specific manipulations of the element can be done.

### Request

1) Please update the return type info from  `HTMLElement` to  `HTMLInputElement` for [widgetCreate_](https://github.com/google/blockly/blob/344861698e65c3fc5f05db000c01cff2155f56ff/core/field_input.ts#L351) in `/core/field_input.ts`! 😁
2) Remove the `as HTMLInputElement` cast in [showInlineEditor](https://github.com/google/blockly/blob/344861698e65c3fc5f05db000c01cff2155f56ff/core/field_input.ts#L334C10-L334C20).

### Alternatives considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by aku1310**
## The basics

- [ ] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

Fixes #7253 

### Proposed Changes

Update the `field_input.ts` file with the following changes:
- Update the return type info from `HTMLElement` to `HTMLInputElement` for [widgetCreate_](https://github.com/google/blockly/blob/344861698e65c3fc5f05db000c01cff2155f56ff/core/field_input.ts#L351)
- Remove the `as HTMLInputElement` cast in [showInlineEditor](https://github.com/google/blockly/blob/344861698e65c3fc5f05db000c01cff2155f56ff/core/field_input.ts#L334C10-L334C20).

### Reason for Changes

Returning proper type info for input-specific manipulations of the element.

### CODE REVIEW

**Comment by aku1310** [comment]
I've signed the CLA after creating the PR. 😅

File: core/field_input.ts [hunk 1]
```diff
@@ -358,7 +358,7 @@ export abstract class FieldInput<T extends InputTypes> extends Field<
       throw new UnattachedFieldError();
     }
     WidgetDiv.show(this, block.RTL, this.widgetDispose_.bind(this));
-    this.htmlInput_ = this.widgetCreate_() as HTMLInputElement;
```
Comment (BeksOmega) [inline core/field_input.ts:361]: This cast is actually still necessary =) So if you could put it back that'd be great!

File: core/field_input.ts [hunk 2]
```diff
@@ -375,7 +375,7 @@ export abstract class FieldInput<T extends InputTypes> extends Field<
    *
    * @returns The newly created text input editor.
    */
-  protected widgetCreate_(): HTMLElement {
+  protected widgetCreate_(): HTMLInputElement {
```
Comment (BeksOmega) [inline core/field_input.ts:378]: This should be `HTMLInputElement | HTMLTextAreaElement`

**Review by BeksOmega — COMMENTED** [review]
Sorry about that! Looks like our issue instructions were slightly incorrect. I should have double checked.

Left comments below about what changes this needs! Thank you for all your work on this :D If it's still running into issues feel free to message me on slack!

File: core/field_input.ts (hunk 1 above)
Comment (aku1310) [reply core/field_input.ts:361]: Okay, will just do that!

**Review by BeksOmega — APPROVED** [review]
Thank you for all of your work on this @aku1310 :D

**Comment by fifi1120** [comment]
hey I am Fiona, glad to join the group! Could you pls assign me some tasks?

**Pull request merged**